src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -36,7 +36,8 @@ public static IResourceBuilder<AzureCosmosDBResource> AddAzureCosmosDB(this IDis
         builder.AddAzureProvisioning();
 
         var resource = new AzureCosmosDBResource(name, ConfigureCosmosDBInfrastructure);
-        return builder.AddResource(resource);
+        return builder.AddResource(resource)
+            .WithAnnotation(new DefaultRoleAssignmentsAnnotation(new HashSet<RoleDefinition>()));
```

Comment:
What does an empty `DefaultRoleAssignmentsAnnotation` actually help us achieve here? This is just here to pass through to `AppliedRoleAssignmentsAnnotation` in the non-ACA scenario, right?

Reply:
> In ACA scenarios, the AzureResourcePreparer will copy the DefaultRoleAssignmentsAnnotation to a regular (empty) RoleAssignmentAnnotation, which will cause the role to be written to the role bicep file.

Ah, I [see where this happens](https://source.dot.net/#Aspire.Hosting.Azure/AzureResourcePreparer.cs,153).

> Thoughts?

IMO, an empty one feels better than a fake role assignment. We clarified in an offline discussion why we need `DefaultRoleAssignments` vs `RoleAssignment` here.